### PR TITLE
InfoMessageFilterFeature: Add local join messages and color to soul point message

### PR DESCRIPTION
**Pull request by kristofbolyai**
image

When in NPC dialogue, local join messages are broken. See #224.

This PR also adds all the missing I18n that was left out (I assume accidentally).

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java [hunk 1]
```diff
@@ -77,6 +78,8 @@ public void onInfoMessage(ChatMessageReceivedEvent e) {
         String msg = e.getCodedMessage();
         MessageType messageType = e.getMessageType();
 
+        System.out.println("msg = " + msg);
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java]: leftover debug I presume.

File: common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java [hunk 2]
```diff
@@ -176,4 +193,21 @@ public void onInfoMessage(ChatMessageReceivedEvent e) {
...
+                            + ChatFormatting.DARK_GREEN + "] " + ChatFormatting.GREEN + playerName);
+            case "VIP+" -> NotificationManager.queueMessage(
+                    ChatFormatting.GREEN + "→ " + ChatFormatting.DARK_AQUA + "[" + ChatFormatting.AQUA + "VIP+"
+                            + ChatFormatting.DARK_AQUA + "] " + ChatFormatting.AQUA + playerName);
+            case "HERO" -> NotificationManager.queueMessage(
+                    ChatFormatting.GREEN + "→ " + ChatFormatting.DARK_PURPLE + "[" + ChatFormatting.LIGHT_PURPLE
+                            + "HERO" + ChatFormatting.DARK_PURPLE + "] " + ChatFormatting.LIGHT_PURPLE + playerName);
+            case "CHAMPION" -> NotificationManager.queueMessage(
+                    ChatFormatting.GREEN + "→ " + ChatFormatting.YELLOW + "[" + ChatFormatting.GOLD + "CHAMPION"
+                            + ChatFormatting.YELLOW + "] " + ChatFormatting.GOLD + playerName);
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java]: This lacks a default branch, and has a decent bit of duplicated code

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java]: No need for default branch, I will fix duplication.

File: common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java]: Yep

File: common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java (hunk 2 above)
Comment (HighCrit) [reply common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java]: We thought so too when HERO was the highest rank. But this shouldn't crash if that changes so 🤷

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java]: Champ branch does not even work, wynn sends hero format for them. Also, we can't just put an universal format.

**Comment by HighCrit** [comment]
Do we want to remove the abilities to hide login announcements with this PR?

**Comment by kristofbolyai** [comment]
> Do we want to remove the abilities to hide login announcements with this PR?

Is there an option to do so in legacy?

**Comment by HighCrit** [comment]
> > Do we want to remove the abilities to hide login announcements with this PR?
> 
> Is there an option to do so in legacy?

Not to my knowledge, but I'd personally prefer to hide instead of redirect it and since it was already in place before this PR we could just leave it in. 

Don't know whether that would be allowed though. As it is one of the rank's perks.

**Comment by magicus** [comment]
Thanks for fixing missing i18n. We should probably have some way to check that we actually have proper i18n keys for all our automated translation stuff, it's so easy to miss when you don't have a literal i18n key in the code.

**Comment by magicus** [comment]
The thought didn't even cross my mind that it'd be considered disallowed to hide login messages. Can we get that verified with Wynncraft staff somehow? @HeyZeer0?

As I think I said before, my original plan was to have an enum `{ KEEP, HIDE, REDIRECT }` for all of these groups, but the PR was large enough as it was. So, unless it is verboten, we should perhaps start by applying this principle to login messages. That way, we won't lose the ability to disable them. (As I'd like to do, personally)

**Comment by kristofbolyai** [comment]
FilterType should be added now.

File: common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java [hunk 3]
```diff
@@ -153,14 +167,23 @@ public void onInfoMessage(ChatMessageReceivedEvent e) {
...
+                if (matcher.find()) {
                     e.setCanceled(true);
+                    if (loginAnnouncements == FilterType.HIDE) {
+                        return;
+                    }
+
+                    String playerName = matcher.group(3);
+                    String rank = matcher.group(2);
+
+                    sendLoginMessage(playerName, rank);
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java:181]: The duplication here is a bit annoying. :( Don't see a very good way to address it, though. Maybe we should make sure the normal and BACKGROUND_ patterns have the same groups (you can always add a dummy `()` to pad it), and they we collect the normal and background patterns into two FilterPattern instances, and select the correct one based on the chat type? 

That sounds like a separate PR, though...

**Review by magicus — APPROVED** [review]

File: common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java (hunk 3 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/InfoMessageFilterFeature.java:181]: Agreed, I think something like was discussed in DC.

**Pull request merged**